.github/workflows/code-climate-coverage-aggregation.yml

```diff
@@ -10,190 +7,316 @@ on:
+                restore-keys: |
+                  ${{ runner.os }}-maven-
+
+            # Run Maven Verify to generate all jacoco reports
+            - name: Build with Maven
+              run: mvn -B verify -P all --file src/backend/pom.xml
+
+            # Loop through all BACKEND services to show that jacoco reports have been generated
+            - name: WHERE AM I - BACKEND?
+              run: |
+                  for s in efiling-backend-demo efiling-api
+                  do
+                      ls ${{ github.workspace }}/src/backend/$s/target/site/jacoco;
+                  done
+              if: ${{ env.ACTION_DEBUG }}
+            
+            # Formatting the BACKEND coverage reports generated (dynamically)
+            - name: Format BACKEND coverage reports
+              run: |
+                  projectRelRegex="^\.\/src\/backend\/(.*)\/target\/site\/jacoco\/jacoco\.xml$"
+                  for file in $(find . -name "jacoco.xml")
+                  do
+                      echo $file
+                      echo $projectRelRegex
+                      if [[ $file =~ $projectRelRegex ]]
+                      then
+                          projectRel="${BASH_REMATCH[1]}"
+                          echo "analyzing project: " $projectRel
+                          projectName="${projectRel//\//-}"
+                          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/backend/$projectRel/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/src/backend/$projectRel/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/$projectName-codeclimate.json;
+                          echo "coverage generated: coverage/$projectName-codeclimate.json;"
+                      else
+                          echo $file does not match
+                      fi
+                  done
+
+            # List all formatted files in coverage directory
+            - name: WHERE AM I - FORMATTED?
+              run: |
+                  ls ${{ github.workspace }}/coverage
+              if: ${{ env.ACTION_DEBUG }}
+            
+            # Count of all total coverage files available
+            - name: Count files present
+              run: |
+                  echo "::set-env name=FILE_COUNTER::$(ls -1q ./coverage | wc -l )"
+                
+            # Sum the coverage reports
+            - name: Summing the coverage reports generated
+              run: ./cc-test-reporter sum-coverage coverage/*-codeclimate.json -p ${{ env.FILE_COUNTER }} -o coverage/total-java-codeclimate.json
+
+            # Upload JSON for debugging purposes
+            - name: Upload JSON for debugging purposes
+              uses: actions/upload-artifact@v2
+              with:
+                  name: summed-java-coverage-report
+                  path: coverage/total-codeclimate.json
+
+    aggregation: 
+
+        name: Aggregate Spring Boot and Javascript reports
+        needs: yarn
```

Comment:
I think this can be formatted like [yarn, sprint-boot] so it waits on both? Mby

Reply:
I can give it a shot, I just went with the longest